x-pack/plugins/canvas/public/components/paginate/index.tsx

```diff
@@ -27,7 +27,7 @@ export const Paginate: React.FunctionComponent<InPaginateProps> = ({
   const initialCurrentPage = totalPages > 0 ? Math.min(startPage, totalPages - 1) : 0;
   const [currentPage, setPage] = useState(initialCurrentPage);
   const hasRenderedRef = useRef<boolean>(false);
-  const maxPage = totalPages - 1;
+  const maxPage = totalPages > 0 ? totalPages - 1 : 0;
```

Comment:
nit: from my point of view` Math.min(totalPages - 1, 0)` looks more semantically right

Reply:
Done.